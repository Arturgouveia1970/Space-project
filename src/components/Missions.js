import React from 'react';

const Missions = () => (
  <div>Missions here</div>
);
export default Missions;
