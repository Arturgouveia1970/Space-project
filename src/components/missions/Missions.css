body {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.missions {
  padding: 3rem 3rem;
}

.missions_table {
  border: 1px solid #ccc;
  margin-top: 70px;
}

.missions_table thead {
  margin-top: 10px;
  border-bottom: 1px solid #ccc;
  height: 30px;
}

.missions_table td,
.missions_table th {
  border-right: 1px solid #ccc;
  padding: 1rem;
}

.missions_table tr:nth-child(even) {
  background-color: #d9d6d6;
}

.missions_table tr:hover {
  background-color: #ddd;
}

.missions th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
}

.mission-name {
  display: flex;
  align-items: flex-start;
  width: 5%;
  font-weight: bolder;
  font-size: 20px;
}

.desc-header {
  border: 1px solid grey;
}

.mission-desc {
  width: 60%;
  line-height: 1.2rem;
  font-size: 12px;
}

button {
  padding: 0.4rem 1rem;
  border: 1px solid black;
}

.status span {
  font-size: 12px;
  background: rgb(69, 72, 65);
  border-radius: 2px;
  color: #fff;
}
