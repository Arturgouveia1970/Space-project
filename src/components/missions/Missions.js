/* eslint-disable camelcase */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux/es/exports';
import { getMissions } from '../../redux/missions/missions';
import './Missions.css';

const Missions = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMissions);
  }, []);

  const missions = useSelector((state) => state.missions);

  return (
    <div className="missions">
      <table className="missions_table">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Join Mission</th>
          </tr>
        </thead>
        {
      missions.map(({
        mission_id, name, description, reserved = false,
      }) => (
        <tr key={mission_id}>
          <td className="mission_name"><h1>{ name }</h1></td>
          <td className="mission_description"><p>{description}</p></td>
          <td>{ reserved ? (<span className="badge badge--primary">ACTIVE mEMBER</span>) : (<span className="badge badge--secondary">NOT A MEMBER</span>) }</td>
          <td>{ reserved ? (<button type="button" className="app-btn-danger">Leave Mission</button>) : (<button type="button" className="app-btn-ghost">Join Mission</button>) }</td>
        </tr>
      ))
    }
      </table>
    </div>
  );
};
export default Missions;
