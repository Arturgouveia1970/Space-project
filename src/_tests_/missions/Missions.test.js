import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Missions from '../../components/missions/Missions';

describe('snapshots testing', () => {
  test('snapshots for Missions component', () => {
    const rendererComponents = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponents).toMatchSnapshot();
  });
});
