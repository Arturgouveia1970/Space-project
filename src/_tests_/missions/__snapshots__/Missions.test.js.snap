// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots testing snapshots for Missions component 1`] = `
<div
  id="missions-section"
  style={
    Object {
      "overflow": "scroll",
    }
  }
>
  <table
    className="missions_table"
  >
    <thead>
      <tr>
        <th
          className="desc-header"
        >
          Mission
        </th>
        <th
          className="desc-header"
        >
          Description
        </th>
        <th
          className="desc-header"
        >
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody
      className="missions"
    />
  </table>
</div>
`;
